packages/@react-aria/table/docs/useTable.mdx

```diff
@@ -488,12 +492,823 @@ function Checkbox(props) {
+element, allowing the user to drag the resizer or use the keyboard arrows to expand the column's width. Be sure to pass the state returned by <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} />
+to this hook so the tracked widths can be updated appropriately. We'll walk through all the changes to the previous table implementation step by step below.
+
+### Table
+
+As mentioned previously, we first need to call <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} /> to initialize and update the widths for our table's columns.
+By providing the hook with `getDefaultMinWidth` and `getDefaultWidth`, we can give our table selection checkbox column a pre-determined width while allowing the other columns divide up the remaining
+width equally initially. We'll pass the state returned by <TypeLink links={statelyDocs.links} type={statelyDocs.exports.useTableColumnResizeState} /> along with any user defined `onResize` handlers
+to our `ResizableTableColumnHeaders` so it can be used by <TypeLink links={docs.links} type={docs.exports.useTableColumnResize} />. The `widths` from this state are provided to each table element so
+the table cells in the body match the columns at all times.
+
+The various style changes below are to make the table itself scrollable and to support table body/column widths greater than the 400px applied to the table itself.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+/*- begin highlight -*/
+import {useTableColumnResizeState} from '@react-stately/table';
+/*- end highlight -*/
+
+function ResizableColumnsTable(props) {
+  let {
+    selectionMode,
+    selectionBehavior,
+    /*- begin highlight -*/
+    onResizeStart,
+    onResize,
+    onResizeEnd
+    /*- end highlight -*/
+  } = props;
+  let state = useTableState({
+    ...props,
+    showSelectionCheckboxes: selectionMode === 'multiple' && selectionBehavior !== 'replace'
+  });
+
+  let ref = useRef();
+  let {collection} = state;
+  let {gridProps} = useTable(
+    {
+      ...props,
+      /*- begin highlight -*/
+      // The table itself is scrollable rather than just the body
+      scrollRef: ref
+      /*- end highlight -*/
+    },
+    state,
+    ref
+  );
+
+  /*- begin highlight -*/
+  let getDefaultWidth = React.useCallback((node) => {
+    // selection cell column should always take up a specific width, doesn't need to be resizable
+    if (node.props.isSelectionCell) {
+      return 20;
+    }
+    return;
+  }, []);
+
+  let getDefaultMinWidth = React.useCallback((node) => {
+     // selection cell column should always take up a specific width, doesn't need to be resizable
+    if (node.props.isSelectionCell) {
+      return 20;
+    }
+    return 75;
+  }, []);
+
+  let layoutState = useTableColumnResizeState({
+    getDefaultWidth,
+    getDefaultMinWidth,
+    tableWidth: 400
```

Comment:
this hard coded width  means the example is a bit limited when it comes to accepting table widths from the user such as `width: auto` or `maxWidth`/`minWidth`, etc. However the alternative is to use a resizeObserver which felt too much for the aria example (and had some finicky behavior will have to poke around)